lingua_franca/format.py

```diff
@@ -483,7 +376,12 @@ def nice_duration(duration, lang=None, speech=True):
     Returns:
         str: timespan as a string
     """
-    if type(duration) is datetime.timedelta:
+    if not lang:
```

Comment:
Once individual languages start localizing it, we can add the decorator. I didn't really work on this, apart from integrating the language-loading business, but it seems to be working in some other languages, as there are Czech tests which seem to pass.

At first glance, it isn't actually operating over English. Instead, it's taking the duration, putting it in the *order* we'd use in English, and assuming that's good enough for the target language. Then it translates each component as it assembles the output.

"~~Two~~ Dos ~~hours~~ horas ~~thirty seven~~ treinta y siete ~~minutes~~ minutos"

Not ideal, but it works for Romance languages, and I'm assuming it works for Czech because there are tests for it in Czech =P when `speech=False`, it just assembles something akin to a digital clock readout with English-appropriate letter abbreviations after `HH:MM:SS` (1d 12:00:00)

Reply:
_Thought I posted this comment previously but can't see it so apologies if this is a duplicate._

Yeah I figured that was the case. Seems like it will work for a lot but not all languages, eg in in Indonesian you'd say the equivalent of "hour two minutes thirty seven". So makes sense as a default and then languages can localize if needed.

Or potentially we could get away with adding a `time_format` to the `res/text/lang/date_time.json`.

Anyway, not for this PR.